tests/Aspire.Hosting.Tests/ResourceExtensionsTests.cs

```diff
@@ -259,6 +259,27 @@ public async Task GetEnvironmentVariableValuesAsyncReturnCorrectVariablesUsingMa
             });
     }
 
+    [Fact]
+    public async Task GetArgumentValuesAsync_ReturnsCorrectValuesForSpecialCases()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+        var surrogate = new ResourceWithConnectionStringSurrogate("ResourceWithConnectionStringSurrogate", _ => "ConnectionString", null);
+        var secretParameter = new ParameterResource("SecretParameter", _ => "SecretParameter", true);
+        var nonSecretParameter = new ParameterResource("NonSecretParameter", _ => "NonSecretParameter");
+
+        builder.AddResource(surrogate);
+        builder.AddResource(secretParameter);
+        builder.AddResource(nonSecretParameter);
+
+        var container = builder.AddContainer("elasticsearch", "library/elasticsearch", "8.14.0")
+            .WithArgs(surrogate)
```

Comment:
We should be passing in the IResourceBuilder<T> here instead of the actual resource itself.

Reply:
yeah you're right, sorry